### ISSUE TITLE
Beginner-friendly error message for misspelled or non-existent gem names

### ISSUE THREAD

**Issue by sealocal** [bundler/bundler#3564]
Let's say I'm new to Ruby, Rails, Bundler, RubyGems, etc.  Clear error messages are a huge help.

Given a Gemfile:
```ruby
source 'https://rubygems.org'
gem 'rails'
gem 'misspelled-gem-name'
```

Bundling will report that the 'misspelled-gem-name' is not available on "this machine":

```console
$ bundle
Fetching gem metadata from https://rubygems.org/........
Fetching version metadata from https://rubygems.org/..
Could not find gem 'misspelled-gem-name (>= 0) ruby' in the gems available on this machine.
```

I think this message could be improved by adding a line that says the gem could not be found in RubyGems, when the source for a Gemfile includes rubygems.org.  For example:

```console
Could not find gem 'misspelled-gem-name (>= 0) ruby' at https://rubygems.org/.
```

Does anyone else think this is would be a welcome improvement?

For further illustration, this spec fails:
```ruby
  describe "failed bundle install" do
    before :each do
      gemfile <<-G
        source 'https://rubygems.org/'
        gem "rails"
        gem "misspelled-gem-name", :group => :development
      G
    end

    it "should report a helpufl error message" do
      bundle :install
      expect(out).to include("Fetching gem metadata from https://rubygems.org/")
      expect(out).to include("Could not find gem 'misspelled-gem-name (>= 0) ruby' at https://rubygems.org/.")
    end
  end
```

```console
Failure/Error: expect(out).to include("Could not find gem 'misspelled-gem-name (>= 0) ruby' at https://rubygems.org/.")
       expected "Fetching gem metadata from https://rubygems.org/...........\nFetching version metadata from https://rubygems.org/..\nCould not find gem 'misspelled-gem-name (>= 0) ruby' in the gems available on this machine." to include "Could not find gem 'misspelled-gem-name (>= 0) ruby' at https://rubygems.org/."
```

**Comment by indirect** [comment]
I think "in any of the sources listed in your Gemfile or installed on this machine" is the most correct error message. How do you feel about that?

**Comment by sealocal** [comment]
I think that's readonable.

**Comment by sealocal** [comment]
Resolved through #3566
